### PR TITLE
insert_tasks_ignore_duplicates - New _insert_tasks_ignore_duplicate_name...

### PR DESCRIPTION
**Pull request by markshaule-wf**
...s to gracefully handle DuplicateTaskNameError exceptions raised by inserts.

@beaulyddon-wf @tannermiller-wf @robertkluin 

FYI: @erikpetersen-wf @jasonaguilon-wf @macleodbroad-wf 

1)  Let me know if you guys would prefer to just modify the existing _insert_tasks function, and add another option.
2)  Should we add an options that just selects the _insert_tasks_ignore_duplicate_names automatically, which shields the implementation from the user.

Right now a typical usage would be:
```
    with context.new(batch_size=100,
                     insert_tasks=insert_tasks_ignore_duplicate_names) as ctx:
      ctx.add(...) 
```

### CODE REVIEW

**Comment by beaulyddon-wf** [comment]
I think it's fine passing in the method like this. One thing we may want to do though is move this new method out of the contexts file and create a module for task insertion handling. That way we can give users a module with some different choices for handling the task insertion without having to dig through Furious internals.

**Comment by markshaule-wf** [comment]
@beaulyddon-wf - Agreed - it doesn't feel right to be using an ```_insert_...``` directly.
Perhaps a new file in ```extras``` package, and we can rename to ```insert_tasks_ignore_duplicate_names```

**Comment by beaulyddon-wf** [comment]
yeah. or `insert_task_handlers` or something to that effect. In case we want to build out other methods. But yeah extras is a good spot for it.

**Comment by markshaule-wf** [comment]
@beaulyddon-wf - Moved new function to furious/extras/insert_task_handlers.py

**Comment by beaulyddon-wf** [comment]
Cool. I like that. +1

**Comment by tylertreat-wf** [comment]
+1

File: furious/tests/extras/test_insert_task_handlers.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from mock import Mock
+from mock import patch
```
Comment (macleodbroad-wf) [inline furious/tests/extras/test_insert_task_handlers.py]: in python 2.7 mock isn't part of the standard library.

File: furious/extras/insert_task_handlers.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        for task in reinsert:
+            try:
+                inserted += _insert_tasks([task], queue, *args, **kwargs)
+            except taskqueue.DuplicateTaskNameError:
+                logging.info(
+                    'Duplicate task name %s detected on queue %s and skipped.'
+                    % (task.name, queue))
+
+        return inserted
```
Comment (macleodbroad-wf) [inline furious/extras/insert_task_handlers.py]: needs blank line at EOF

**Comment by macleodbroad-wf** [comment]
+1

File: furious/extras/insert_task_handlers.py (hunk 2 above)
Comment (tannermiller-wf) [inline furious/extras/insert_task_handlers.py]: Yes, another True Believer of the Ending Blank Line.

**Comment by tannermiller-wf** [comment]
+1

File: furious/tests/extras/test_insert_task_handlers.py (hunk 1 above)
Comment (markshaule-wf) [inline furious/tests/extras/test_insert_task_handlers.py]: Alright - I'll move down and group with google.

File: furious/extras/insert_task_handlers.py (hunk 2 above)
Comment (markshaule-wf) [inline furious/extras/insert_task_handlers.py]: corrected.  I also had an extra blank line at the end of the test file which was removed.

**Comment by markshaule-wf** [comment]
@macleodbroad-wf - comments addressed regarding imports and blank lines.

**Comment by macleodbroad-wf** [comment]
+1

**Comment by tannermiller-wf** [comment]
+1

**Comment by macleodbroad-wf** [comment]
+1

**Pull request merged**